lib/common/http1client.c

```diff
@@ -595,6 +610,17 @@ static void on_whole_request_sent(h2o_socket_t *sock, const char *err)
     }
 }
 
+static void on_header_sent(h2o_socket_t *sock, const char *err)
+{
+    struct st_h2o_http1client_t *client = sock->data;
+
+    h2o_timer_unlink(&client->super._timeout);
```

Comment:
Are we fine with just disabling the timer?

I get that we do not need to ream it when an error has occurred, but unless that is the case, I think we should be arming either an I/O timeout or a first-byte timeout?

Reply:
Partially addressed in https://github.com/h2o/h2o/pull/3316/commits/5770af1e4f79482ae58e5eacdb43eba3dc5d6da5
I came up with one TODO I added as a comment. If the response status of stream is `HEADER`, or it's not `HEADER` but the received response headers include `expect` header, there must be no issues. Otherwise, i.e. http1client already receives response header but it didn't contain `expect` header, waiting 100 means stall. Should we let it stall or return error to the client? Or start sending req body regardless of the absence of expect header? I appreciate your suggestion.

Reply:
Very good question! When receiving responses to Expect: 100-continue, I think what we want to do is:
* if the response is 100, start sending the request body
* if the response is 1xx other than 100, continue waiting for 100, while forwarding 1xx
* if the response is final (i.e, other than 100 or 1xx), start sending the request body.

Regarding the last bullet point, note that there's nothing that prohibits servers start sending a final response before receiving a complete request. RFC 9112 (HTTP/1.1) might be silent about this, but [RFC 9113 section 8.1](https://httpwg.org/specs/rfc9113.html#HttpFraming) explicitly states that "a server can send a complete response prior to the client sending an entire request if the response does not depend on any portion of the request that has not been sent and received."

Under such situation, while we might not need to send the entire request, I think it might make sense for us to try sending the entire request until the server closes the connection, recalling the design choices that we have made in the past for handling these early responses in our server-side code.

Reply:
Thank you for the suggestion. https://github.com/h2o/h2o/pull/3316/commits/4dade945ae45cdd215772b4e107622fe0c4e621f makes h2o start sending request body when it receives final response.